prow/github/client.go

```diff
@@ -187,15 +190,17 @@ func (c *Client) Throttle(hourlyTokens, burst int) {
 //   An endpoint is used when all preceding endpoints have returned a conn err.
 //   This should be used when using the ghproxy GitHub proxy cache to allow
 //   this client to bypass the cache if it is temporarily unavailable.
-func NewClient(token string, bases ...string) *Client {
+func NewClient(token string, secretPath string, secretAgent *config.SecretAgent, bases ...string) *Client {
```

Comment:
This API is a bit unclear -- you give the token as read once, then the agent and the path. I don't think it's necessary for the GitHub client to need to know about the inner workings of the secret agent, so maybe we can instead pass `func() (string)` that could generate the secret at run-time as necessary:

```go
func whatever() {
    agent := NewSecretAgent(tokens)
    client := NewClient(func() string {
        return agent.Config()[githubToken]
    }, bases...)
}

func NewClient(token func() string, bases ...string) *Client {
    ...
}
```

Reply:
This is the largest point of feedback